integration/test_ml.py

```diff
@@ -0,0 +1,389 @@
+    existing_model1 = _ensure_model_exists(name_only_model)
+    existing_model2 = _ensure_model_exists(name_and_tags_model)
+    filter_str = 'displayName={0} OR tags:{1}'.format(
+        existing_model1.display_name, existing_model2.tags[0])
+
+    models_list = ml.list_models(list_filter=filter_str)
+    assert len(models_list.models) == 2
+    for mdl in models_list.models:
+        assert mdl == existing_model1 or mdl == existing_model2
+    assert models_list.models[0] != models_list.models[1]
+
+    _clean_up_model(existing_model1)
+    _clean_up_model(existing_model2)
+
+
+def test_list_models_invalid_filter():
+    invalid_filter = 'InvalidFilterParam=123'
+
+    with pytest.raises(exceptions.InvalidArgumentError) as excinfo:
+        ml.list_models(list_filter=invalid_filter)
+    check_firebase_error(excinfo, 400, 'Request contains an invalid argument.')
+
+
+def test_delete_model(name_only_model):
+    existing_model = _ensure_model_exists(name_only_model)
+
+    ml.delete_model(existing_model.model_id)
+
+    # Second delete of same model will fail
+    with pytest.raises(exceptions.NotFoundError) as excinfo:
+        ml.delete_model(existing_model.model_id)
+    check_firebase_error(excinfo, 404, 'Requested entity was not found.')
+
+
+#'pip install tensorflow' in the environment if you want _TF_ENABLED = True
+#'pip install tensorflow=2.0.0' for version 2 etc.
+if _TF_ENABLED:
+    # Test tensor flow conversion functions if tensor flow is enabled.
+    SAVED_MODEL_DIR = '/tmp/saved_model/1'
+
+    def _clean_up_tmp_directory():
+        if os.path.exists(SAVED_MODEL_DIR):
+            shutil.rmtree(SAVED_MODEL_DIR)
+
+    @pytest.fixture
+    def keras_model():
+        x_array = [-1, 0, 1, 2, 3, 4]
+        y_array = [-3, -1, 1, 3, 5, 7]
+        model = tf.keras.models.Sequential(
+            [tf.keras.layers.Dense(units=1, input_shape=[1])])
+        model.compile(optimizer='sgd', loss='mean_squared_error')
+        model.fit(x_array, y_array, epochs=3)
+        yield model
+
+    @pytest.fixture
+    def saved_model_dir(keras_model):
+        # different versions have different model conversion capability
+        # pick something that works for each version
+        save_dir = SAVED_MODEL_DIR
+        _clean_up_tmp_directory() # previous failures may leave files
+        if tf.version.VERSION.startswith('1.'):
+            tf.reset_default_graph()
+            x_var = tf.placeholder(tf.float32, (None, 3), name="x")
+            y_var = tf.multiply(x_var, x_var, name="y")
+            with tf.Session() as sess:
+                tf.saved_model.simple_save(sess, save_dir, {"x": x_var}, {"y": y_var})
+        else:
+            # If it's not version 1.x or version 2.x we need to update the test.
+            assert tf.version.VERSION.startswith('2.')
+            tf.saved_model.save(keras_model, save_dir)
+        yield save_dir
+
+
+    def test_from_keras_model(keras_model):
```

Comment:
Lets remove the `if` condition at the module-level, and just add an `@skipif` decorator on the TF-aware tests.

Reply:
removed the if. replaced with skipUnless to avoid the not.